src/utils/response/rawApiResponse.ts

```diff
@@ -18,22 +20,52 @@ export type RawApiResponse = {
   ErrorMessage?: string;
   ErrorInternal?: string;
   ErrorExtended?: unknown;
+  _ignoreTypeGuard?: boolean; // workaround until SP-351 is done, we don't want to update every test by hand
 };
 
-export async function createRawApiResponse(promise: Promise<AxiosResponse>): Promise<RawApiResponse> {
-  const axiosResponse: AxiosResponse = await promise;
-
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  if (axiosResponse === undefined) {
-    throw new Error('The Api response is unsuccessful');
+function handleResponse(axiosResponse: AxiosResponse): RawApiResponse {
+  if (isRawApiResponse(axiosResponse.data) && axiosResponse.data.Success) {
+    axiosResponse.data._ignoreTypeGuard = undefined;
+    return axiosResponse.data;
   }
+  throw axiosResponse;
+}
 
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-  const rawApiResponse: RawApiResponse = axiosResponse.data;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function handleError(error: any) {
+  const apiResponseError: TimeTacApiError = {
+    reason: ErrorReason.ReponseFailed,
+    _plainError: JSON.stringify(error),
+    response: undefined,
+  };
 
-  if (!rawApiResponse.Success) {
-    throw new Error('The Api response is unsuccessful');
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  if (error.data !== undefined) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    apiResponseError.response = error.data;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  if (Boolean(error.response) && Boolean(error.response.data)) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    apiResponseError.response = error.response.data;
   }
```

Comment:
I tried adding a type, for now, that reflects what's checked in the if so that we don't have to add the eslint disables: 
```
function handleError(error: { data?: unknown; response?: { data?: unknown } }) {
  const apiResponseError: TimeTacApiError = {
    reason: ErrorReason.ReponseFailed,
    _plainError: JSON.stringify(error),
    response: undefined,
  };

  if (error.data !== undefined) {
    apiResponseError.response = error.data;
  }
  if (error.response?.data != null) {
    apiResponseError.response = error.response.data;
  }

  return Promise.reject(apiResponseError);
}
```
Maybe `{ data?: unknown} | { response?: { data?: unknown } }` would make more sense?

Reply:
Done with { data?: unknown} | { response?: { data?: unknown } }